### ISSUE TITLE
yolov8p6

### ISSUE THREAD

**Issue by Ellohiye** [ultralytics/ultralytics#22507]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

For images of size 4096*4096, does the YOLOv8P61280 model perform better than the YOLOv8S640? Is there any benefit of the YOLOv8S640 over the YOLOv8P61280 in high-resolution image detection, other than time?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Ellohiye, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (e.g., comparing YOLOv8 variants on high-resolution images), please provide as much information as possible to help us give targeted guidance:
-Your task and dataset details (object sizes, density, sample images)
-Exact commands or scripts used (including image sizes, batch size, augmentations)
-Training/evaluation logs and metrics (mAP, recall, inference speed)
-Hardware and environment (GPU model, CUDA/PyTorch versions)
-Confirmation you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
It will perform better if the objects are large. When you increase image size, you make objects larger, so you need a deeper feature like P6 for larger receptive field.

**Comment by Ellohiye** [comment]
Wouldn't the effect be even better for small objects? The P6 series magnifies the image, making small objects appear larger.

**Comment by glenn-jocher** [comment]
Good question—more pixels help small objects, but that comes from a larger `imgsz`, not from P6 itself; P6 mainly adds a stride-64 head and deeper backbone for very large objects, while small-object recall is driven by the P3 (stride-8) head, so a P5 model at higher `imgsz` (or tiling) typically matches or beats P6 for small/dense targets at lower cost; if you’re starting fresh, YOLO11 is a better default than YOLOv8 at the same sizes as shown in the [YOLO11 vs YOLOv8 comparison](https://docs.ultralytics.com/compare/yolo11-vs-yolov8/).

**Comment by Ellohiye** [comment]
Does this mean that YOLOv8P6 offers improvements for both small targets (image size 1280) and large targets (P6 layer)? Is the YOLOv8P6 model more suitable for target detection of different sizes?

**Comment by glenn-jocher** [comment]
Not exactly—P6 is not a universal upgrade; use it when your dataset has many very large objects, while for small/dense or mixed‑scale scenes a higher‑resolution P5 (ideally YOLO11) plus optional tiled inference is usually stronger; see the SAHI slicing integration and the YOLO11 model docs for guidance: [SAHI slicing integration](https://docs.ultralytics.com/guides/sahi-tiled-inference/), [YOLO11 model](https://docs.ultralytics.com/models/yolo11/).

**Comment by Ellohiye** [comment]
If the P5 series 1280 increases the input image size, will the detection performance for medium and large targets improve?

**Comment by glenn-jocher** [comment]
Yes—bumping a P5 model to `imgsz=1280` typically lifts AP on medium/large objects; most of that gain comes from the higher resolution, while P6 mainly helps when objects are extremely large; for best results, train or fine-tune at 1280 and verify with COCO APm/APl via:

```bash
yolo val model=path/to/weights.pt data=path/to/data.yaml imgsz=1280 save_json=True
```

See the Val mode options in the [Ultralytics validation docs](https://docs.ultralytics.com/modes/val/).

**Comment by Ellohiye** [comment]
But wouldn't increasing the resolution increase the detection rate of small targets

**Comment by glenn-jocher** [comment]
Yes—raising `imgsz` typically boosts small‑object recall, but gains taper once the smallest targets still cover only a few pixels; train and infer at the larger size, or use tiling if they remain tiny—quick test with a P5 model at 1280 (ideally YOLO11):
```bash
yolo train model=yolo11s.pt data=path/to/data.yaml imgsz=1280
```
See training args in the [Train mode docs](https://docs.ultralytics.com/modes/train/).
